### PR TITLE
Addition of deployment scripts

### PR DESCRIPTION
**Pull request by NexFlare**
The commit add the ability to fully automate publishing of GitHub Releases and updating of the GitHub Pages index to include the newly published charts.

This commit has following changes:

- Addition of relman submodule - A release management tool
- publish-chart.yaml - Runs the entire workflow which includes
      1. Checkout code
      2. Package helm chart
      3. Remove uncommitted changes   
      4. Publish GitHub release
      5. Update the release index in [gh-pages branch](https://github.com/nuodb/nuodb-insights/tree/gh-pages)

The example of that can be seen in workflow of forked repo: https://github.com/NexFlare/nuodb-insights/actions/runs/6200573459/job/16835712296

This was triggered by creating a changelog and tagging the commit as v2.0.0 as follows:
```sh
./release/relman.py --check-current --create-changelog --commit 
git tag v2.0.0
git push --tags
```
This created a GitHub release for [v2.0.0](https://github.com/NexFlare/nuodb-insights/releases/tag/v2.0.0) and generated a commit to the [gh-pages branch](https://github.com/NexFlare/nuodb-insights/commit/a39623568fae12e093e685195009e8165db9c158) that updates the index to add the new charts.

For reference about relman visit https://github.com/nuodb/release-tools/tree/main

### CODE REVIEW

**Comment by adriansuarez** [comment]
@NexFlare, can you please add a description of the changes to the summary of the PR, and make sure that they appear in the commit message once this is merged into master?

**Comment by adriansuarez** [comment]
@NexFlare, you created a fork of this repo that you used to exercise these changes. Can you please include a link the workflow run and generated commit in your fork, and also some commands that prove that it works?

I'm looking for similar information to what I posted here: https://github.com/nuodb/nuodb-helm-charts/pull/321#issuecomment-1694723857

File: scripts/remove-uncommited-changes.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+set -e
+GIT_STATUS="$(git status --procelain)"
```
Comment (adriansuarez) [inline scripts/remove-uncommited-changes.sh:4]: Misspelled `--porcelain`.

File: scripts/remove-uncommited-changes.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+VERSION="$(./release/relman.py --check-current --show-head-version)"
```
Comment (adriansuarez) [inline scripts/remove-uncommited-changes.sh:5]: Why is this needed?

File: scripts/remove-uncommited-changes.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+if [ "$GIT_STATUS" != "" ]
+then
+    git checkout -f -- "./stable/insights/Chart.lock"
```
Comment (adriansuarez) [inline scripts/remove-uncommited-changes.sh:8]: You just want to remove any uncommitted changes to this particular file. Why don't you just execute this unconditionally? The end result is the same, which is that there are no uncommitted changes to this file. You can probably just remove this script and execute this command directly in the workflow steps.

File: .github/workflows/publish-chart.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+          fi
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+      - name: Package Helm charts
+        run: ./scripts/package.sh
+      - name: Remove uncommitted changes
+        run: ./scripts/remove-uncommited-changes.sh
+      - name: Publish GitHub release
+        run: |
+          gh release create v${{ steps.vars.outputs.version }} \
+            -t "NuoDB Helm Charts ${{ steps.vars.outputs.version }}" \
```
Comment (sivanov-nuodb) [inline .github/workflows/publish-chart.yaml:39]: This needs to be updated to "NuoDB Insights v${{ steps.vars.outputs.version }}"

**Comment by sivanov-nuodb** [comment]
@NexFlare Are you planning to update the README.md with the new Helm repository URL for NuoDB Insights as part of this PR?

File: scripts/package.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+fail() {
+    printf "$1\n" >&2
+    exit 1
+}
+
+package() {
+    # Create directory for packaged Helm charts
+    dest="package/$1/v$(./get-version.sh)"
```
[the remark this answers is not in the document]
Comment (NexFlare) [reply scripts/package.sh:10]: `**/*.tgz` is already added in gitignore file

File: scripts/remove-uncommited-changes.sh [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+set -e
+
+git checkout -f -- "./stable/insights/Chart.lock"
```
[the remark this answers is not in the document]
Comment (NexFlare) [reply scripts/remove-uncommited-changes.sh:7]: Added the required changes.

**Comment by NexFlare** [comment]
> @NexFlare Are you planning to update the README.md with the new Helm repository URL for NuoDB Insights as part of this PR?

Changed the url in `stable/README.md`

File: scripts/package.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (sivanov-nuodb) [reply scripts/package.sh:10]: It might be used by other scripts that prepare the `index.yaml`. Although the index file is moved to the target place, it would be good to explicitly ignore the `package` directory.

**Review by sivanov-nuodb — APPROVED** [review]
Looks good.
